src/H5Ppublic.h

```diff
@@ -5733,52 +5733,194 @@ H5_DLL herr_t H5Pset_virtual_view(hid_t dapl_id, H5D_vds_view_t view);
- *          \p expression is NULL, \p size is ignored and the function
- *          does not write anything to the buffer. The function always
- *          returns the size of the data transform expression.
+ *          If \p expression is non-NULL, up to \p size bytes of the data transform expression are written to
+ * the buffer. If \p expression is NULL, \p size is ignored, and the function does not write anything to the
+ *          buffer. The function always returns the size of the data transform expression.
  *
- *          If 0 is returned for the size of the expression, no data
- *          transform expression exists for the property list.
+ *          If 0 is returned for the size of the expression, no data transform expression exists for the
+ * property list.
  *
- *          If an error occurs, the buffer pointed to by \p expression is
- *          unchanged and the function returns a negative value.
+ *          If an error occurs, the buffer pointed to by \p expression is unchanged, and the function returns
+ * a negative value.
+ *
+ * \par Example
+ *      An example snippet from examples/h5_dtransform.c:
+ *      \snippet h5_dtransform.c H5Pget_data_transform_snip
  *
  * \since 1.8.0
  *
  */
-H5_DLL ssize_t   H5Pget_data_transform(hid_t plist_id, char *expression /*out*/, size_t size);
+H5_DLL ssize_t H5Pget_data_transform(hid_t plist_id, char *expression /*out*/, size_t size);
+/**
+ *
+ * \ingroup  DXPL
+ *
+ * \brief Determines whether error-detection is enabled for dataset reads
+ *
+ * \param[in]  plist_id Dataset transfer property list identifier
+ *
+ * \return Returns \p H5Z_ENABLE_EDC or \p H5Z_DISABLE_EDC if successful;
+ *         otherwise returns a negative value.
+ *
+ * \details H5Pget_edc_check() queries the dataset transfer property
+ *          list \p plist to determine whether error detection is enabled for
+ *          data read operations.
+ *
+ * \since 1.6.0
+ *
+ */
 H5_DLL H5Z_EDC_t H5Pget_edc_check(hid_t plist_id);
-H5_DLL herr_t    H5Pget_hyper_vector_size(hid_t fapl_id, size_t *size /*out*/);
-H5_DLL int       H5Pget_preserve(hid_t plist_id);
-H5_DLL herr_t    H5Pget_type_conv_cb(hid_t dxpl_id, H5T_conv_except_func_t *op, void **operate_data);
-H5_DLL herr_t    H5Pget_vlen_mem_manager(hid_t plist_id, H5MM_allocate_t *alloc_func, void **alloc_info,
-                                         H5MM_free_t *free_func, void **free_info);
-H5_DLL herr_t    H5Pset_btree_ratios(hid_t plist_id, double left, double middle, double right);
-H5_DLL herr_t    H5Pset_buffer(hid_t plist_id, size_t size, void *tconv, void *bkg);
+/**
+ *
+ * \ingroup  DXPL
+ *
+ * \brief Retrieves number of I/O vectors to be read/written in hyperslab I/O
+ *
+ * \param[in]   fapl_id Dataset transfer property list identifier
+ * \param[out]  size    Number of I/O vectors to accumulate in memory for I/O operations
+ *
+ * \return Returns a non-negative value if successful; otherwise returns a negative value.
```

Comment:
Can you replace all `herr_t` returns with `\return \herr_t`? That way we prevent repetition and have to change it only in one place should we ever change our mind.

Reply:
done